ui/components/app/confirm-data/confirm-data.js

```diff
@@ -0,0 +1,72 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { useSelector } from 'react-redux';
+
+import {
+  Color,
+  TextVariant,
+  TEXT_TRANSFORM,
+} from '../../../helpers/constants/design-system';
+import { getKnownMethodData } from '../../../selectors';
+import { useI18nContext } from '../../../hooks/useI18nContext';
+import { useTransactionFunctionType } from '../../../hooks/useTransactionFunctionType';
+
+import Box from '../../ui/box/box';
+import Disclosure from '../../ui/disclosure';
+import TransactionDecoding from '../transaction-decoding';
+import { Text } from '../../component-library';
+
+const ConfirmData = ({ txData, dataComponent }) => {
+  const t = useI18nContext();
+  const { txParams = {} } = txData;
+  const methodData = useSelector(
+    (state) => getKnownMethodData(state, txParams.data) || {},
+  );
+  const { functionType } = useTransactionFunctionType(txData);
+
+  if (dataComponent) {
+    return dataComponent;
+  }
+
+  if (!txParams.data) {
+    return null;
```

Comment:
I'm seeing blank tabs when I issue a transaction

<img width="759" alt="Screen Shot 2023-02-28 at 7 04 19 PM" src="https://user-images.githubusercontent.com/20778143/221850473-c2020434-3863-48ff-be40-82da726729c0.png">
<img width="764" alt="Screen Shot 2023-02-28 at 7 04 30 PM" src="https://user-images.githubusercontent.com/20778143/221850491-fe11d109-4310-462b-b179-a0b034909739.png">
<img width="1680" alt="Screen Shot 2023-02-28 at 7 09 52 PM" src="https://user-images.githubusercontent.com/20778143/221850821-af6aa029-66de-43d0-9db3-2a5142f70c5e.png">

Reply:
@digiwand : plz test some contract interaction.

Also, you can compare with code before this PR to know if data should be visible.

Reply:
@jpuri 

I tested on develop. We should be hiding the tabs if there is no data or hex data to show. e.g. "Send Legacy Transaction" and "Send EIP 1559 Transaction" should not show the tabs. This branch is displaying the tabs

Reply:
oh I see now. `renderData` and `renderDataHex` could return `null` in the past. They are now returning the react elements

Reply:
Thanks a lot @digiwand , very good catch. You saved regression 😄 

I updated the PR 🙏